### DISCUSSION TITLE
PROLIFIC_PID is undefined

### DISCUSSION THREAD

**Discussion by aislinnkeogh** [jspsych/jsPsych#2959] [Q&A] [answered]
As the title suggests, I'm having an issue with Prolific integration.

I'm doing what I've always successfully done before: telling Prolific I'll use URL parameters to record Prolific IDs, and using the following line in my code to do that:

`var prolific_id = jsPsych.data.getURLVariable('PROLIFIC_PID');`

But I've just run a little pilot study and `prolific_id` is undefined for all my participants. Has there been a change in Prolific's API or something?

**Comment by jodeleeuw** [comment, marked as answer]
I just checked Prolific and it looks like nothing has changed on their end. 

Is it possible that you are using the variable before it is defined?

**Reply by aislinnkeogh** [reply]
I don't *think* that can be the case: the only time it should be being used is when I call my data saving function in the `on_finish` of all my trials, which should only happen when the timeline runs, right? It's also defined in the same place as a random ID number which is making it into the data no problem.

**Reply by jodeleeuw** [reply]
I just noticed in some recent Prolific studies from my students that the URL parameters weren't actually showing up in the URL. In the Prolific site, so you see the `?PROLIFIC_ID={{PROLIFIC_ID}}&...` added to your study URL? If not, change the selection to something else and then select the URL parameters again.

**Reply by aislinnkeogh** [reply]
I *just* noticed this today and was about to let you know that the problem was one of user (i.e. me) error, not jsPsych error :) thanks for your help!
